compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt

```diff
@@ -402,6 +405,49 @@ internal fun ModuleDescriptor.findClassOrTypeAlias(
   return null
 }
 
+internal fun KtClassOrObject.typeVariableNames(
+  module: ModuleDescriptor
+): List<TypeVariableName> {
+  // Any type which is constrained in a `where` clause is also defined as a type parameter.
+  // It's also technically possible to have one constraint in the type parameter spot, like this:
+  // class MyClass<T : Any> where T : Set<*>, T : MutableCollection<*>
+  // Merge both groups of type parameters in order to get the full list of bounds.
+  val boundsByVariableName = mutableMapOf<String, MutableList<TypeName>>()
+
+  typeParameterList
+    ?.parameters
+    ?.forEach { parameter ->
+
+      if (parameter.fqNameOrNull(module) == null) {
```

Comment:
That if check could be a `.filter {..}`

Reply:
This is funny because I'd considered it after I made the PR, then decided I was nitpicking and should just leave it alone.  I should have gone with my first instinct. :)